### ISSUE TITLE
Implement better detection of "multipart/encrypted" email with PGP encrypted message

### ISSUE THREAD

**Issue by martgil** [FlowCrypt/flowcrypt-ios#2580]
Description: The FlowCrypt iOS didn't recognize "multipart/encrypted" email with PGP encrypted message. Emails at https://mail.google.com/mail/u/flowcrypt.compatibility@gmail.com/#inbox/FMfcgzGxTFXMFnDNbgknpqvXKtSmMgSK isn't recognized by the FlowCrypt app and shows not encrypted + no signed blank email thread.

Reference: 
- https://github.com/FlowCrypt/flowcrypt-browser/pull/5727
- https://github.com/FlowCrypt/flowcrypt-android/blob/e58d040f8eade675ca8c6bd18cb608d46bcdb44b/FlowCrypt/src/main/java/com/flowcrypt/email/extensions/jakarta/mail/PartExt.kt#L82

**Comment by ioanmo226** [comment]
@sosnovsky I think better to finish this task first and then release current 1.3.1 version?

**Comment by sosnovsky** [comment]
Yes, after fixing this one we can publish a new release to App Store and switch back to `flowcrypt-browser` tasks.

**Comment by ioanmo226** [comment]
@sosnovsky I don't have task assigned for iOS at the moment. Do I have to move on browser extension?

**Comment by ioanmo226** [comment]
Okay, I see
I thought Mart was already working on it.

**Comment by ioanmo226** [comment]
@martgil I can reproduce this issue in iOS app with email flowcrypt.compatibility account.
However could you guide me how to reproduce such email for UI testing?
Thank you

**Comment by martgil** [comment]
Hi @ioanmo226, I'll try if I can craft one for you. Reason is that, the particular test email we have was just sent by the user and haven't had the chance to know to generate one for ourselves that time.

**Comment by martgil** [comment]
Hello @ioanmo226 - I can't seemed to craft a test email for this one easily, may I ask for a little more time.

**Comment by ioanmo226** [comment]
I see. Thank you
No problem
I haven't even implemented a fix yet.

**Comment by martgil** [comment]
Hello @ioanmo226, there's some bad news here. I'm unable to generate such email even after making such similar mime email. 

I'm using the python (3.11.9) code below, while it sends similar mime email and being recognizable by FlowCrypt, the decryption fails:

```python
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.base import MIMEBase
from email import encoders

# Email account credentials
sender_email = "user@gmail.com"
recipient_email = "user@gmail.com"
password = "app specific password" # this canbe obtained in https://myaccount.google.com/apppasswords

boundary = "----=_Part_7_39145192.1715228097793"
msg = MIMEMultipart("encrypted", boundary=boundary)
msg["From"] = sender_email
msg["To"] = recipient_email
msg["Subject"] = "test multipart/encrypted"
msg["MIME-Version"] = "1.0"  

pgp_encrypted_part = MIMEBase("application", "pgp-encrypted")
pgp_encrypted_part.add_header("Content-Transfer-Encoding", "7bit")
pgp_encrypted_part.set_payload("Version: 1")

# The email.txt contains ascii armored pgp encrypted email similar to the sample email in flowcrypt.compatibility
with open("email.txt", "r") as file:
    encrypted_message = file.read()

pgp_data_part = MIMEBase("application", "octet-stream")
pgp_data_part.add_header("Content-Transfer-Encoding", "7bit")
pgp_data_part.set_payload(encrypted_message)

msg.attach(pgp_encrypted_part) 
msg.attach(pgp_data_part)       

# Send the email via Gmail's SMTP server
with smtplib.SMTP_SSL("smtp.gmail.com", 465) as server:
    server.login(sender_email, password)
    server.sendmail(sender_email, recipient_email, msg.as_string())

print("Email sent successfully.")
```

**Comment by ioanmo226** [comment]
You mean that test email was recognized by Flowcrypt which means Flowcrypt knows it is PGP but can't decrypt that email?

**Comment by martgil** [comment]
Yes, that's what is happening in the test email generated using the python script when opened through the FlowCrypt browser extension.

**Comment by martgil** [comment]
If I remember it correctly, what I did there was just to replace the sensitive infos like exact email subject and sender while retaining the format for a "multipart/encrypted" mime email.
